### ISSUE TITLE
New goreleaser workflow forces maintainers to have it installed

### ISSUE THREAD

**Issue by khos2ow** [mattermost/mattermost-mattermod#251]
With addition of goreleaser workflow, now the maintainers have to have the binary installed and `GITHUB_TOKEN` exported while cutting a release tag. This is, I think, mostly because of the `before.hooks` defined in it.
/cc @streamer45

_Issue created from a [Mattermost message](https://community-release.mattermost.com/_redirect/pl/9ubfyobdb3feijrrefhhiby9so) *by @metanerd*._

**Comment by streamer45** [comment]
Yes, the token should be passed for `goreleaser` to be able to publish the release.

**Comment by khos2ow** [comment]
This might be a bit inconvenient for the person who cuts the release and potentially inconsistent in the github release page. What I think we should do is to move this release workflow to a remote CI and not on a maintainers laptops.

**Comment by streamer45** [comment]
Sure. The whole process can be done in response to pushing a new version tag. We just need to implement something like [this](https://goreleaser.com/ci/actions/). I can open a ticket if you agree.

**Comment by khos2ow** [comment]
I'm completely fine with this, @metanerd @jaydeland what do you think of this suggestion?

**Comment by metanerd** [comment]
Wait, we would use circleci to lint/test, github actions to cut the release, gitlab to deploy?
Is this a pressing matter? I would wait a bit for maybe another solution.

**Comment by jaydeland** [comment]
Yes - can we move this CI to GitLab before including this enhancement?

> Wait, we would use circleci to lint/test, github actions to cut the release, gitlab to deploy?
> Is this a pressing matter? I would wait a bit for maybe another solution.

**Comment by streamer45** [comment]
> Wait, we would use circleci to lint/test, github actions to cut the release, gitlab to deploy?
> Is this a pressing matter? I would wait a bit for maybe another solution.

Actually, Github actions was just an example. Goreleaser supports all of those CIs so if we want to converge to one that also works.

**Comment by metanerd** [comment]
Ah ok, sorry, I totally didn't get it. So we should migrate our pipeline to gitlab.
